### PR TITLE
201 logic app xslt with params

### PR DESCRIPTION
**Pull request by vinaysingh123**
Logic App quick start template - Creates a request-response Logic Apps which perform XSLT based transformation using parameters.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vinaysingh123__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (vinaysin). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by vinaysingh123** [comment]
@gatneil  can you please review this?

**Comment by vinaysingh123** [comment]
@bmoore-msft  can you please review this?

File: 201-logic-app-xslt-with-params/metadata.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "itemDisplayName": "Azure Logic Apps - XSLT with parameters",
+    "description": "creates a request-response Logic App which performs XSLT based transformation. The XSLT map takes primitives (integer, string etc.) as input parameters as uses them during XML transformation.",
```
Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/metadata.json]: use sentence casing

File: 201-logic-app-xslt-with-params/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-logic-app-xslt-with-params%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-logic-app-with-params%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template creates a request-response Logic App which performs XSLT based transformation. The XSLT map takes primitives (integer, string etc.) as input parameters as uses them during XML transformation.
+`Tags: Logic Apps, Enterprise Integration, XSLT, XML Transform
```
Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/README.md]: move tags to bottom

File: 201-logic-app-xslt-with-params/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Prerequisites ##
+
+In order to properly deploy this ARM template, you need to first create an Azure Function app and provide its name in the parameters.
+Create an Azure Function App - https://portal.azure.com/#create/Microsoft.FunctionApp
+
+## Deployment steps
+
+You can click the "deploy to Azure" button at the beginning of this document or follow the instructions for command line deployment using the scripts in the root of this repo.
```
Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/README.md]: "Deploy to Azure" (capitalization)

File: 201-logic-app-xslt-with-params/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
...-16\"?><xsl:stylesheet xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\" xmlns:input=\"urn:my-input-variables\" xmlns:msxsl=\"urn:schemas-microsoft-com:xslt\" xmlns:var=\"http://schemas.microsoft.com/BizTalk/2003/var\" exclude-result-prefixes=\"msxsl var s0 userCSharp\" version=\"1.0\" xmlns:ns0=\"http://Integration.Destination\" xmlns:s0=\"http://Integration.Source\" xmlns:userCSharp=\"http://schemas.microsoft.com/BizTalk/2003/userCSharp\"><xsl:import href=\"https://az818438.vo.msecnd.net/functoids/functoidsscript.xslt\" /><xsl:output omit-xml-declaration=\"yes\" method=\"xml\" version=\"1.0\" />  <xsl:param name=\"input:X\"></xsl:param><xsl:param name=\"input:Y\"></xsl:param><xsl:template match=\"/\"><xsl:apply-templates select=\"/s0:Root\" /></xsl:template><xsl:template match=\"/s0:Root\"><ns0:Root><Sum><xsl:value-of select=\"userCSharp:MathAdd($input:X, $input:Y)\" /></Sum><Diff><xsl:value-of select=\"userCSharp:MathSubtract($input:X, $input:Y)\" /></Diff><Product><xsl:value-of select=\"userCSharp:MathMultiply($input:X, $input:Y)\" /></Product><Division><xsl:value-of select=\"userCSharp:MathDivide($input:X, $input:Y)\" /></Division></ns0:Root></xsl:template></xsl:stylesheet>",
```
Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: should this be hardcoded?  https://az818438.vo.msecnd.net/functoids/functoidsscript.xslt

Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: use https wherever possible (schemas.microsoft.com)

File: 201-logic-app-xslt-with-params/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                  "Succeeded"
+                ]
+              },
+              "type": "Response"
+            },
+            "Transform_XML": {
+              "inputs": {
+                "content": "@{triggerBody()}",
+                "function": {
+                  "id": "[concat(resourceId('microsoft.web/sites/', parameters('existingFunctionAppName')), '/functions/', parameters('transformFunctionName'))]"
```
Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: just use resourceId('Microsoft.Web/sites/functions', parameters('existingFunctionAppName'), parameters('transformFunctionName')

File: 201-logic-app-xslt-with-params/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+              "type": "httpTrigger"
+            },
+            {
+              "name": "res",
+              "direction": "out",
+              "type": "http"
+            }
+          ]
+        },
+        "files": {
```
Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: The inline xslt is really hard to digest - can this be broken out into a file (i.e. does the RP support that?)  It's unlikely a human could replicate your feat...

File: 201-logic-app-xslt-with-params/artifacts/sample-input.xml [hunk 7]
```diff
@@ -0,0 +1 @@
+﻿<ns0:Root xmlns:ns0="http://Integration.Source" />
```
Comment (bmoore-msft) [inline 201-logic-app-xslt-with-params/artifacts/sample-input.xml]: what are these files for?  (this one and the xsl?)

File: 201-logic-app-xslt-with-params/azuredeploy.json (hunk 4 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: Yes, this is a fixed CDN site url.

File: 201-logic-app-xslt-with-params/artifacts/sample-input.xml (hunk 7 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/artifacts/sample-input.xml]: These are sample input payloads that I generally put under artifacts folder. Users can use this input data to fire the logic app.

File: 201-logic-app-xslt-with-params/metadata.json (hunk 1 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/metadata.json]: Done.

File: 201-logic-app-xslt-with-params/README.md (hunk 3 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/README.md]: Done.

File: 201-logic-app-xslt-with-params/azuredeploy.json (hunk 4 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: Fixed.

File: 201-logic-app-xslt-with-params/azuredeploy.json (hunk 6 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: This isn't possible today. I will fix it once we support alternate mechanisms in the RP (hopefully later this year).

File: 201-logic-app-xslt-with-params/azuredeploy.json (hunk 5 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: Fixed.

File: 201-logic-app-xslt-with-params/azuredeploy.json (hunk 4 above)
Comment (vinaysingh123) [inline 201-logic-app-xslt-with-params/azuredeploy.json]: Fixed at most of the places. Left it in few places where it breaks the app.

**Comment by vinaysingh123** [comment]
@bmoore-msft  I have made changes based on your feedback. Can you please pull the changes?

**Pull request merged**